### PR TITLE
Setup CI release

### PR DESCRIPTION
**Pull request by armanbilge**
Who could argue that sbt-gh-actions is preferable to this!! :P

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -17,6 +17,14 @@ val commonSettings = Seq(
   scalacOptions ++= Seq("-deprecation", "-feature", "-Xfatal-warnings")
 )
 
+val noPublishSettings = Seq(
+  skip in publish := true,
```
Comment (japgolly) [inline build.sbt:21]: I believe in 1.5.5 this is all we need, all the rest is redundant stuff that I created years ago in the 0.13.x days, and now it seems to have spread around verbatim haha. Plus change to `publish / skip`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    strategy:
+      matrix:
+        scalajsversion: ["1.x", "0.6.x"]
+    env:
+      SCALAJS_VERSION: "${{ matrix.scalajsversion == '0.6.x' && '0.6.28' || '' }}"
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+      - uses: olafurpg/setup-scala@v10
```
Comment (japgolly) [inline .github/workflows/release.yml:17]: 13 is latest. I'll add dependabot for these later (or feel free to yourself if you want)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        scalajsversion: ["1.x", "0.6.x"]
+    env:
+      SCALAJS_VERSION: "${{ matrix.scalajsversion == '0.6.x' && '0.6.28' || '' }}"
+    steps:
+      - uses: actions/checkout@v2.3.4
```
Comment (japgolly) [inline .github/workflows/release.yml:14]: Just `@v2`, that way it will pick up compatible 2.x.x actions automatically.

**Review by japgolly — CHANGES_REQUESTED** [review]
> Who could argue that sbt-gh-actions is preferable to this!! :P

Hehe well not me :D

**Comment by armanbilge** [comment]
Btw do you want to publish snapshots?

File: .github/workflows/release.yml (hunk 2 above)
Comment (armanbilge) [reply .github/workflows/release.yml]: Never used dependabot. My chance to learn! This is what you install? https://github.com/apps/dependabot-preview

**Review by japgolly — CHANGES_REQUESTED** [review]
I was about to merge and then I thought I'd just do a quick comparison against boopickle and they've got this before the publish step:

```yml
      - uses: actions/checkout@v2
      - uses: olafurpg/setup-scala@v12
      - uses: olafurpg/setup-gpg@v3
      - uses: coursier/cache-action@v6
```

I'm not fussed about the caching but can you pls also confirm whether we need `setup-gpg`?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+name: Release
+on:
+  push:
+    tags: ["v*"]
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
```
Comment (japgolly) [inline .github/workflows/release.yml:7]: `ubuntu-latest` probably better

**Comment by armanbilge** [comment]
> I'm not fussed about the caching but can you pls also confirm whether we need setup-gpg?

Oops, you are right we don't. I remember you advised me to check with boopickle, my bad. Let me do that now. I think I copied this directly from the sbt-ci-release repo ...

**Comment by armanbilge** [comment]
See https://github.com/olafurpg/sbt-ci-release/pull/171 about `setup-pgp`

**Review by japgolly — APPROVED** [review]

**Comment by japgolly** [comment]
Done! Thanks and nice work!

**Comment by japgolly** [comment]
@armanbilge The release-drafter stuff broke. It didn't run on this PR so we missed it

https://github.com/scala-js/scala-js-dom/actions/runs/1122419282

**Comment by armanbilge** [comment]
Yeah I'm looking into it now. I copied it straight out of boopickle so not sure what's missing. Btw, we published our first snapshot! So seems that's working at least :)

**Comment by armanbilge** [comment]
@japgolly ok, I don't have a clue sorry :( it says we are missing the `template` property from the release-drafter ocnfig but it's definitely there ... I really did just copy this from boopickle. Any ideas?

**Comment by armanbilge** [comment]
@japgolly found it https://github.com/release-drafter/release-drafter/issues/398#issuecomment-617928416

In short, the `release-drafter.yml` config has to be on the master branch. So it should fix itself when we next merge series/1.x into master.

**Comment by japgolly** [comment]
@armanbilge np disabled on `series/1.x` in 014eedd8650d157570a08da11cf5d9579daba7ad

**Pull request merged**